pkg/orm/task.go

```diff
@@ -25,6 +26,65 @@ func NewTaskORM() *TaskORM {
 	return &TaskORM{dbClient: clientWrapper.Client, clientWrapper: clientWrapper}
 }
 
+// Define cache keys as constants similar to rate limiter pattern
+type TaskCacheKey string
+
+const (
+	TaskByIdCacheKey      TaskCacheKey = "task"
+	TasksByWorkerCacheKey TaskCacheKey = "tasks_by_worker"
+)
+
+type TaskCache struct {
+	key      TaskCacheKey
+	id       string
+	workerId string
+	offset   int
+	limit    int
+	types    []db.TaskType
+}
+
+func NewTaskCache(key TaskCacheKey) *TaskCache {
+	return &TaskCache{
+		key: key,
+	}
+}
+
+func (tc *TaskCache) GetCacheKey() string {
+	switch tc.key {
+	case TaskByIdCacheKey:
+		return fmt.Sprintf("%s:%s", tc.key, tc.id)
+	case TasksByWorkerCacheKey:
+		// Create a shorter hash for task types
+		typeHash := ""
+		if len(tc.types) > 0 {
+			typeStrs := make([]string, len(tc.types))
+			for i, t := range tc.types {
+				typeStrs[i] = string(t)[0:2] // Take first 2 chars of each type
```

Comment:
what if we have this is first 3 chars right? what happens if we have task type text_to_image and text_to_three_d? will be hard to differentiate between them right